crates/db_schema/src/aggregates/structs.rs

```diff
@@ -100,6 +100,8 @@ pub struct PostAggregates {
   pub community_id: CommunityId,
   pub creator_id: PersonId,
   pub controversy_rank: f64,
+  /// A rank that amplifies smaller communities
+  pub scaled_rank: i32,
```

Comment:
That's true, these all should probably be made into floats (not sure why I didn't want to use floats originally), could you open an issue for that? Lets do that as a separate issue.

Also being a "news" type sort, the time decay is supposed to go to zero after ~ 2 days, after which published descending takes over.

- https://join-lemmy.org/docs/contributors/07-ranking-algo.html
- https://medium.com/hacking-and-gonzo/how-hacker-news-ranking-algorithm-works-1d9b0cf2c08d

Reply:
Mh, that's interesting. Then maybe we can just keep hot_rank as is but make the new scaled_rank a float? I think it would make sense to make the new scaled_rank function and column float from the start instead of migration later. 

There's no code change required for making this a float but keeping hot_rank an int

Reply:
Maybe I should just change hot_rank as a part of this tbh, since scaled_rank depends on it.

Reply:
Sounds good. Note it will reduce the performance of the scheduled task a diary bit since afterwards almost no post will have a hot rank of 0 and be filtered out (ideas @sunaurus? )

Reply:
That's def a concern... We could either use the published timestamp as a filter, or minimum hot_rank threshold.

Reply:
I haven't had a chance to look at the implementation yet, so sorry if this is a dumb idea, but can we not just update the rank functions to set the rank 0 after it decays to some threshold, to guarantee that the majority of posts will still have a 0 value for rank and thus get filtered out during updates?

Reply:
I mean something like if the rank is below 1.0 (or perhaps even higher!) then just set it to 0

Reply:
Yes, that should work. But it would need careful consideration what the threshold should be both for hot_rank and scaled_rank. According to dess above, the published sort is supposed to take over after a few days. So maybe it would be better to make the function return zero if published is more than 7 days in the past? The hot_rank function gets `published` as a parameter anyways.

So that change would just be to replace `IF (hours_diff > 0) THEN` with `IF (hours_diff > 0 AND hours_diff < 24 * 7) THEN`